src/pattern.py

```diff
@@ -14,9 +14,9 @@
 
 for i in reversed(range(1, n)):
     for j in reversed(range(1, i+1)):
-        if j != i:
+        if j != 1:
             pattern += "* "
-        elif j == i:
+        elif j == 1:

```

Comment:
An `else` would work here. You don't need to check both if `j != 1` and `j == 1`; it has to be one or the other.